integrations/llama_index/src/ucai_llamaindex/toolkit.py

```diff
@@ -129,28 +114,39 @@ def uc_function_to_llama_tool(
             raise ValueError("Either function_name or function_info should be provided.")
 
         fn_schema = generate_function_input_params_schema(function_info)
+        pydantic_model = fn_schema.pydantic_model
+
+        # Enforce strict validation by setting 'extra' to 'forbid' for both outer and inner models
+        WrappedModel = create_model(
```

Comment:
I'm not the biggest fan of this extreme restriction. I feel like this is going to be an annoying maintenance burden if LlamaIndex modifies the internal data structures again.